Updates to KSPM Integration README (#3964)
elastic/integrations: packages/cloud_security_posture/docs/README.md

# Kubernetes Security Posture Management 

This integration periodically monitors and compares Kubernetes infrastructure against security best practices defined by CIS to help security, DevOps, and DevSecOps personnel to: 

1. Identify and remediate misconfigurations 

> 2. Understand the overall security posture of their Kubernetes clusters both- individually and holistically 

Reviewer: ```suggestion
2. Understand the overall security posture of their Kubernetes clusters, both individually and holistically 
```

## Integration Assets 

> After this integration has been installed for the first time, the following assets will get created and made available in the Security solution UI: 

Reviewer: ```suggestion
After this integration has been installed for the first time, the following assets will get created and made available in the Elastic Security UI: 
```

| Asset             | Description                                                                                                                                         |
| ----------------- | --------------------------------------------------------------------------------------------------------------------------------------------------- |

> | Posture Dashboard | The posture dashboard provides an overview of the security posture of all Kubernetes clusters monitored                                                |

Reviewer: ```suggestion
| Cloud Posture dashboard | The Cloud Posture dashboard provides an overview of the security posture of all monitored Kubernetes clusters                                                |
```

> | Findings          | Findings communicate the outcome of a specific resource being evaluated with a specific rule. All latest findings are viewable on the findings page |

Reviewer: ```suggestion
| Findings          | Findings communicate whether resources are compliant with applicable rules. Current findings are viewable on the Findings page |
```

> | Benchmark Rules   | Benchmark rules are used to assess Kubernetes resources for secure configuration. Benchmark rules are viewable on the Benchmark page                                                                                                                                                   |

Reviewer: ```suggestion
| Benchmark Rules   | Benchmark rules are used to assess Kubernetes resources for secure configuration. Benchmark rules are viewable on the Benchmarks page                                                                                                                                                   |
```

## Compatibility

This integration is tested with Kubernetes 1.21.x and currently supports the security posture assessment of:

1. [Unmanaged/Vanilla Kubernetes clusters](https://kubernetes.io/)
2. [Amazon EKS clusters](https://aws.amazon.com/eks/)

This integration has not been tested on 

1. Amazon EKS on AWS Outposts

This Integration does not currently support the security posture assessment of:

1. Google GKE clusters
2. Azure AKS
3. Red Hat Openshift 
4. Amazon EKS with AWS Fargate nodes

## Permissions 

> This integration requires access to node files, node processes, and the Kubernetes api-server therefore it assumes the agent will be installed as a [DaemonSet](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/) with the proper [Roles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#role-and-clusterrole) and [RoleBindings](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#rolebinding-and-clusterrolebinding) attached.

Reviewer: ```suggestion
This integration requires access to node files, node processes, and the Kubernetes api-server. Therefore it assumes the agent will be installed as a [DaemonSet](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/) with the proper [Roles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#role-and-clusterrole) and [RoleBindings](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#rolebinding-and-clusterrolebinding) attached.
```

If deploying this integration on an [Amazon EKS cluster](https://docs.aws.amazon.com/eks/latest/userguide/what-is-eks.html), specific AWS permissions are required for the IAM user to make specific AWS API calls. To enable the integration to collect configuration metadata from all relevant resources/services, make sure these permissions are given:

> - ecr:GetRegistryPolicy

Reviewer 2: ```suggestion
```yaml
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "VisualEditor0",
            "Effect": "Allow",
            "Action": [
                "ecr:GetRegistryPolicy",
                "eks:ListTagsForResource",
                "elasticloadbalancing:DescribeTags",
                "ecr-public:DescribeRegistries",
                "ecr:DescribeRegistry",
                "elasticloadbalancing:DescribeLoadBalancerPolicyTypes",
                "ecr:ListImages",
                "ecr-public:GetRepositoryPolicy",
                "elasticloadbalancing:DescribeLoadBalancerAttributes",
                "elasticloadbalancing:DescribeLoadBalancers",
                "ecr-public:DescribeRepositories",
                "eks:DescribeNodegroup",
                "ecr:DescribeImages",
                "elasticloadbalancing:DescribeLoadBalancerPolicies",
                "ecr:DescribeRepositories",
                "eks:DescribeCluster",
                "eks:ListClusters",
                "elasticloadbalancing:DescribeInstanceHealth",
                "ecr:GetRepositoryPolicy"
            ],
            "Resource": "*"
        }
    ]
}
```
```

> 
> - ecr:GetRegistryPolicy
> - eks:ListTagsForResource
> - elasticloadbalancing:DescribeTags
> - ecr-public:DescribeRegistries
> - ecr:DescribeRegistry
> - elasticloadbalancing:DescribeLoadBalancerPolicyTypes
> - ecr:ListImages
> - ecr-public:GetRepositoryPolicy
> - elasticloadbalancing:DescribeLoadBalancerAttributes
> - elasticloadbalancing:DescribeLoadBalancers
> - ecr-public:DescribeRepositories
> - eks:DescribeNodegroup
> - ecr:DescribeImages
> - elasticloadbalancing:DescribeLoadBalancerPolicies
> - ecr:DescribeRepositories
> - eks:DescribeCluster
> - eks:ListClusters
> - elasticloadbalancing:DescribeInstanceHealth
> - ecr:GetRepositoryPolicy

Reviewer 2: ```suggestion
```suggestion
```yaml
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "VisualEditor0",
            "Effect": "Allow",
            "Action": [
                "ecr:GetRegistryPolicy",
                "eks:ListTagsForResource",
                "elasticloadbalancing:DescribeTags",
                "ecr-public:DescribeRegistries",
                "ecr:DescribeRegistry",
                "elasticloadbalancing:DescribeLoadBalancerPolicyTypes",
                "ecr:ListImages",
                "ecr-public:GetRepositoryPolicy",
                "elasticloadbalancing:DescribeLoadBalancerAttributes",
                "elasticloadbalancing:DescribeLoadBalancers",
                "ecr-public:DescribeRepositories",
                "eks:DescribeNodegroup",
                "ecr:DescribeImages",
                "elasticloadbalancing:DescribeLoadBalancerPolicies",
                "ecr:DescribeRepositories",
                "eks:DescribeCluster",
                "eks:ListClusters",
                "elasticloadbalancing:DescribeInstanceHealth",
                "ecr:GetRepositoryPolicy"
            ],
            "Resource": "*"
        }
    ]
}
```
``````

## Leader election

> To collect cluster level data (compared to node level information) the integration makes use of the [leader election](https://www.elastic.co/guide/en/fleet/master/kubernetes_leaderelection-provider.html) mechanism.

Reviewer: ```suggestion
To collect cluster level data (rather than just node level information) the integration makes use of the [leader election](https://www.elastic.co/guide/en/fleet/master/kubernetes_leaderelection-provider.html) mechanism.
```
